### PR TITLE
Fix HttpClient.detach usages

### PR DESCRIPTION
**Pull request by vogdb**
```
json:
timestamp: 2016-04-06T12:27:54+00:00
message: Unexpected rejection error.
hostname: janus2.fuckbook.com
level: error
error:
message: http-client error: No such session 1942071661
stack:
Error: http-client error: No such session 1942071661
    at /usr/lib/node_modules/cm-janus/lib/janus/http-client.js:36:15
    at tryCatcher (/usr/lib/node_modules/cm-janus/node_modules/request-promise/node_modules/bluebird/js/main/util.js:26:23)
    at Promise._settlePromiseFromHandler (/usr/lib/node_modules/cm-janus/node_modules/request-promise/node_modules/bluebird/js/main/promise.js:507:31)
    at Promise._settlePromiseAt (/usr/lib/node_modules/cm-janus/node_modules/request-promise/node_modules/bluebird/js/main/promise.js:581:18)
    at Promise._settlePromises (/usr/lib/node_modules/cm-janus/node_modules/request-promise/node_modules/bluebird/js/main/promise.js:697:14)
    at Async._drainQueue (/usr/lib/node_modules/cm-janus/node_modules/request-promise/node_modules/bluebird/js/main/async.js:123:16)
    at Async._drainQueues (/usr/lib/node_modules/cm-janus/node_modules/request-promise/node_modules/bluebird/js/main/async.js:133:10)
    at Immediate.Async.drainQueues [as _onImmediate] (/usr/lib/node_modules/cm-janus/node_modules/request-promise/node_modules/bluebird/js/main/async.js:15:14)
    at processImmediate [as _immediateCallback] (timers.js:367:17)
```

### CODE REVIEW

**Comment by vogdb** [comment]
@tomaszdurka please review

File: lib/janus/plugin/streaming.js [hunk 1]
```diff
@@ -54,7 +54,9 @@ PluginStreaming.prototype.subscribe = function() {
     });
   }))
     .catch(JanusError.Error, function(error) {
-      serviceLocator.get('http-client').detach(plugin);
+      serviceLocator.get('http-client').detach(plugin).catch(function() {
+        serviceLocator.get('logger').warn('Failed detach of plugin', plugin.getContext());
```
Comment (tomaszdurka) [inline lib/janus/plugin/streaming.js]: Please add error context.

File: lib/janus/plugin/streaming.js [hunk 2]
```diff
@@ -71,7 +73,9 @@ PluginStreaming.prototype.publish = function() {
...
```
Comment (tomaszdurka) [inline lib/janus/plugin/streaming.js]: Same goes here.

File: test/spec/janus/plugin/streaming.js [hunk 3]
```diff
@@ -61,6 +61,7 @@ describe('PluginStreaming', function() {
     context('on unsuccessful subscribe', function() {
       beforeEach(function() {
         streams.addSubscribe.restore();
+        httpClient.detach.returns(Promise.resolve());
```
Comment (tomaszdurka) [inline test/spec/janus/plugin/streaming.js]: So it was popping with unhandled rejection before?

Comment (vogdb) [inline test/spec/janus/plugin/streaming.js]: yes. Check this test
```js
Promise.reject(new Error('Hiii'))
  .then(function() {
  })
  .catch(function() {
    Promise.reject(new Error('Nooooooo'))/*.catch(function(){
      console.log('Sneaky one!');
    })*/;
    throw new Error('Byyyyyyyeeee');
  })
  .catch(function(){
    console.log('Got it');
  });
```

**Comment by vogdb** [comment]
@tomaszdurka please review

**Comment by tomaszdurka** [comment]
:lgtm:

**Pull request merged**